includes/frontend/class-wcmp-cart-fees.php

```diff
@@ -41,26 +41,35 @@ public function __construct()
      */
     public function get_delivery_options_fees(WC_Cart $cart): void
     {
-        if (is_admin() && ! defined('DOING_AJAX')) {
+        if (! defined('DOING_AJAX') && is_admin()) {
             return;
         }
 
         if (isset($_POST['post_data'])) {
             // non-default post data for AJAX calls
-            parse_str($_POST['post_data'], $post_data);
+            parse_str($_POST['post_data'], $postData);
         } else {
             // checkout finalization
-            $post_data = $_POST;
+            $postData = $_POST;
         }
 
-        /*  check for delivery options & add fees*/
-        if (empty($post_data[WCMP_Admin::META_DELIVERY_OPTIONS])) {
+        // Check if delivery options exist at all.
+        if (empty($postData[WCMP_Admin::META_DELIVERY_OPTIONS])) {
             return;
         }
 
-        $delivery_options_data = $post_data[WCMP_Admin::META_DELIVERY_OPTIONS];
-        $delivery_options_data = json_decode(stripslashes($delivery_options_data), true);
-        $this->deliveryOptions = DeliveryOptionsAdapterFactory::create($delivery_options_data);
+        $deliveryOptionsData = $postData[WCMP_Admin::META_DELIVERY_OPTIONS];
+        $deliveryOptionsData = json_decode(stripslashes($deliveryOptionsData), true);
+
+        /*
+         * Check if delivery options is null. Happens when when switching to a shipping method that does not allow
+         * showing delivery options, for example.
+         */
+        if (!$deliveryOptionsData) {
```

Comment:
```suggestion
        if (! $deliveryOptionsData) {
```
https://wiki.dmp.zone/pages/viewpage.action?spaceKey=MD&title=DSR-1%3A+PHP+Standards#DSR-1:PHPStandards-Operators(includingcasting)needtohaveawhitespace